File: rough_work/process_midori.py
import os

def merge_seqs(output):
    pass


def process_midori(input_path,
                   output_path,
                   keep_species = True,
                   latin_binomial = None):
    copy_input(input_path, output_path)
    merge_seqs(output_path)
    if assign_taxonomy:
        process_taxa(output_path, latin_binomial)
    else:
        pass


def merge_seqs(fpath):
    cmd = r"""cat {fpath} | awk '/^>/ {{printf("\n%s\n",$0);next; }} {{ printf("%s",$0);}}  END {{printf("\n");}}' > test3.fasta
    """.format(fpath=fpath)
    print(cmd)
    os.system(cmd)

def copy_input(input_path, output_path):
    cmd = "cp {} {}".format(input_path, output_path)
    os.system(cmd)

output_path = "/Users/krissankaran/Desktop/test_output.fasta"
merge_seqs(output_path)

File: rough_work/process_midori.R


merge_seqs <- function(output)


process_midori <- function(input_path,
                           output_path,
                           keep_species = TRUE,
                           latin_binomial = NULL) {
  copy_input(input_path, output_path)
  merge_seqs(output_path)
  if (assign_taxonomy) {
    process_taxa(output_path, latin_binomial)
  } else {

  }
}

merge_seqs <- function(fpath) {

  cmd <- paste0(
    fpath,
    ' | awk "',
    '/^>/ {printf("\n%s\n",$0);next; }',
    '{ printf("%s",$0);}  END {printf("\n");}"')

  cmd <- paste0(
    fpath,
    " | awk \'/^>/ {printf("\n%s\n\",$0);next; } { printf(\"%s\",$0);}  END {printf(\"\n\");}'"
  )
    system(cmd)
}

copy_input <- function(input_path, output_path) {
  cmd <- sprintf("cp %s %s", input_path, output_path)
  system(cmd)
}
  